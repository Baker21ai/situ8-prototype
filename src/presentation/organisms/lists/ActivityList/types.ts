/**
 * TypeScript interfaces for ActivityList compound component
 */

import { EnterpriseActivity, ActivityCluster } from '../../../../../lib/types/activity';
import { Priority } from '../../../../../lib/utils/status';
import { EnterpriseCardVariant } from '../../../../../components/EnterpriseActivityCard';

// Filter state for activities
export interface ActivityFilterState {
  search: string;
  priorities: Priority[];
  statuses: string[];
  types: string[];
  locations: string[];
  timeRange: 'live' | '15m' | '1h' | '4h' | '24h' | 'custom';
  businessImpact: string[];
  confidenceThreshold: number;
  showClusters: boolean;
  aiFiltering: boolean;
}

// Performance metrics
export interface ActivityPerformanceMetrics {
  totalActivities: number;
  renderTime: number;
  filteredCount: number;
  criticalCount: number;
  averageConfidence: number;
  processingRate: number;
}

// Layout modes
export type ActivityLayoutMode = 'grid' | 'horizontal';
export type ActivityViewMode = 'stream' | 'minimal' | 'summary';

// Context data for compound component
export interface ActivityListContextData {
  // Data
  activities: (EnterpriseActivity | ActivityCluster)[];
  filteredActivities: (EnterpriseActivity | ActivityCluster)[];
  
  // State
  filters: ActivityFilterState;
  selectedItems: Set<string>;
  viewMode: ActivityViewMode;
  layoutMode: ActivityLayoutMode;
  compactMode: boolean;
  useVirtualScrolling: boolean;
  showPerformanceMetrics: boolean;
  
  // Metrics
  performanceMetrics: ActivityPerformanceMetrics;
  priorityMetrics: Record<Priority, number>;
  
  // Event handlers
  onActivitySelect?: (activity: EnterpriseActivity | ActivityCluster) => void;
  onActivityAction?: (action: string, activity: EnterpriseActivity | ActivityCluster) => void;
  onBulkAction?: (action: string, activities: EnterpriseActivity[]) => void;
  
  // State setters
  setFilters: (filters: Partial<ActivityFilterState>) => void;
  setSelectedItems: (items: Set<string>) => void;
  setViewMode: (mode: ActivityViewMode) => void;
  setLayoutMode: (mode: ActivityLayoutMode) => void;
  setCompactMode: (compact: boolean) => void;
  setUseVirtualScrolling: (enabled: boolean) => void;
  setShowPerformanceMetrics: (show: boolean) => void;
  
  // Derived data
  hasSelection: boolean;
  isEmpty: boolean;
  
  // Configuration
  realTimeMode?: boolean;
  height?: number;
  className?: string;
}

// Component props
export interface ActivityListProps {
  activities: (EnterpriseActivity | ActivityCluster)[];
  onActivitySelect?: (activity: EnterpriseActivity | ActivityCluster) => void;
  onActivityAction?: (action: string, activity: EnterpriseActivity | ActivityCluster) => void;
  onBulkAction?: (action: string, activities: EnterpriseActivity[]) => void;
  realTimeMode?: boolean;
  className?: string;
  height?: number;
  children?: React.ReactNode;
}

// Header component props
export interface ActivityListHeaderProps {
  children?: React.ReactNode;
  className?: string;
}

// Content component props
export interface ActivityListContentProps {
  className?: string;
  height?: number;
}

// Footer component props
export interface ActivityListFooterProps {
  className?: string;
}

// Filters component props
export interface ActivityListFiltersProps {
  className?: string;
  showAdvanced?: boolean;
}

// Stats component props
export interface ActivityListStatsProps {
  className?: string;
  detailed?: boolean;
}

// Search component props
export interface ActivityListSearchProps {
  placeholder?: string;
  className?: string;
}

// View toggle component props
export interface ActivityListViewToggleProps {
  className?: string;
}

// Priority segment for display
export interface PrioritySegment {
  priority: Priority;
  items: (EnterpriseActivity | ActivityCluster)[];
  count: number;
  color: {
    bg: string;
    border: string;
    text: string;
  };
  icon: React.ReactNode;
}

// Compound component type with sub-components
export interface ActivityListCompound {
  (props: ActivityListProps): JSX.Element;
  Header: React.ComponentType<ActivityListHeaderProps>;
  Content: React.ComponentType<ActivityListContentProps>;
  Footer: React.ComponentType<ActivityListFooterProps>;
  Filters: React.ComponentType<ActivityListFiltersProps>;
  Stats: React.ComponentType<ActivityListStatsProps>;
  Search: React.ComponentType<ActivityListSearchProps>;
  ViewToggle: React.ComponentType<ActivityListViewToggleProps>;
}