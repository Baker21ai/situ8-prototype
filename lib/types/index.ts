/**
 * Central export for all type definitions
 */

// Re-export all activity types
export * from './activity';

// Re-export all guard types
export * from './guards';

// Re-export all communication types
export * from './communications';

// Re-export utility types
export type { Priority, Status, BusinessImpact } from '../utils/status';
export type { ActivityType, ThreatLevel, SecurityLevel } from '../utils/security';

// Common UI component prop types
export interface BaseComponentProps {
  className?: string;
  children?: React.ReactNode;
}

export interface InteractiveComponentProps extends BaseComponentProps {
  onClick?: () => void;
  onHover?: () => void;
  disabled?: boolean;
}

// Component size variants
export type ComponentSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';

// Component variants
export type ComponentVariant = 'default' | 'outline' | 'ghost' | 'destructive' | 'success';

// Layout variants
export type LayoutVariant = 'compact' | 'comfortable' | 'spacious';

// View modes
export type ViewMode = 'grid' | 'list' | 'table' | 'kanban' | 'timeline';

// Sort options
export interface SortOption<T = string> {
  field: T;
  direction: 'asc' | 'desc';
  label: string;
}

// Pagination
export interface PaginationState {
  page: number;
  pageSize: number;
  total: number;
  totalPages: number;
}

// Selection state
export interface SelectionState<T = string> {
  selected: Set<T>;
  lastSelected?: T;
  mode: 'single' | 'multiple';
}

// Filter state
export interface FilterState {
  active: boolean;
  filters: Record<string, any>;
  quickFilters?: string[];
}

// Search state
export interface SearchState {
  query: string;
  isSearching: boolean;
  results?: any[];
  resultCount?: number;
}

// Modal/Dialog props
export interface ModalProps extends BaseComponentProps {
  isOpen: boolean;
  onClose: () => void;
  title?: string;
  description?: string;
  size?: 'sm' | 'md' | 'lg' | 'xl' | 'full';
}

// Form field props
export interface FormFieldProps<T = string> {
  name: string;
  label?: string;
  value: T;
  onChange: (value: T) => void;
  error?: string;
  required?: boolean;
  disabled?: boolean;
  placeholder?: string;
}

// Action handler types
export type ActionHandler<T = void> = (data: T) => void | Promise<void>;
export type SelectHandler<T> = (item: T) => void;
export type MultiSelectHandler<T> = (items: T[]) => void;

// API response types
export interface ApiResponse<T> {
  data: T;
  success: boolean;
  message?: string;
  error?: string;
  timestamp: Date;
}

export interface PaginatedResponse<T> extends ApiResponse<T[]> {
  pagination: PaginationState;
}

// Error types
export interface AppError {
  code: string;
  message: string;
  details?: Record<string, any>;
  timestamp: Date;
}

// Notification types
export interface Notification {
  id: string;
  type: 'info' | 'success' | 'warning' | 'error';
  title: string;
  message?: string;
  timestamp: Date;
  duration?: number;
  action?: {
    label: string;
    handler: () => void;
  };
}