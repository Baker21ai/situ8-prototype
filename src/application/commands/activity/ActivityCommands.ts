/**
 * Activity Domain Commands
 * All commands related to activity management
 */

import { ICommand } from '../base/ICommand';
import { ActivityType } from '../../../lib/utils/security';
import { Priority, Status } from '../../../lib/utils/status';

// ===== CREATE ACTIVITY COMMAND =====

export interface CreateActivityCommand extends ICommand {
  readonly type: 'CreateActivity';
  readonly data: {
    activityType: ActivityType;
    title: string;
    location: string;
    priority: Priority;
    description?: string;
    building?: string;
    zone?: string;
    assignedTo?: string;
    confidence?: number;
    externalData?: {
      sourceSystem: string;
      originalType: string;
      rawPayload: Record<string, any>;
    };
  };
}

export interface CreateActivityResult {
  activityId: string;
  autoTagsApplied: string[];
  incidentTriggered?: {
    incidentId: string;
    rule: string;
  };
}

// ===== UPDATE ACTIVITY COMMAND =====

export interface UpdateActivityCommand extends ICommand {
  readonly type: 'UpdateActivity';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    updates: {
      priority?: Priority;
      status?: Status;
      description?: string;
      assignedTo?: string;
      userTags?: string[];
    };
    reason?: string;
  };
}

export interface UpdateActivityResult {
  updatedFields: string[];
  triggeredWorkflows?: string[];
}

// ===== ASSIGN ACTIVITY COMMAND =====

export interface AssignActivityCommand extends ICommand {
  readonly type: 'AssignActivity';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    assignedTo: string;
    assignedBy: string;
    reason?: string;
    notifyAssignee?: boolean;
  };
}

export interface AssignActivityResult {
  previousAssignee?: string;
  notificationSent: boolean;
}

// ===== ARCHIVE ACTIVITY COMMAND =====

export interface ArchiveActivityCommand extends ICommand {
  readonly type: 'ArchiveActivity';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    reason: string;
    archivedBy: string;
    permanent?: boolean;
  };
}

export interface ArchiveActivityResult {
  archivedAt: Date;
  retentionDate: Date;
}

// ===== BULK STATUS UPDATE COMMAND =====

export interface BulkUpdateStatusCommand extends ICommand {
  readonly type: 'BulkUpdateStatus';
  readonly data: {
    activityIds: string[];
    status: Status;
    reason?: string;
    notifyAssigned?: boolean;
  };
}

export interface BulkUpdateStatusResult {
  successCount: number;
  failedIds: string[];
  notificationsSent: number;
}

// ===== ADD USER TAG COMMAND =====

export interface AddUserTagCommand extends ICommand {
  readonly type: 'AddUserTag';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    tag: string;
    addedBy: string;
  };
}

export interface AddUserTagResult {
  tag: string;
  allTags: string[];
}

// ===== REMOVE USER TAG COMMAND =====

export interface RemoveUserTagCommand extends ICommand {
  readonly type: 'RemoveUserTag';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    tag: string;
    removedBy: string;
  };
}

export interface RemoveUserTagResult {
  tag: string;
  allTags: string[];
}

// ===== ESCALATE ACTIVITY COMMAND =====

export interface EscalateActivityCommand extends ICommand {
  readonly type: 'EscalateActivity';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    escalationType: 'priority' | 'assignment' | 'incident';
    targetPriority?: Priority;
    targetAssignee?: string;
    createIncident?: boolean;
    reason: string;
    escalatedBy: string;
  };
}

export interface EscalateActivityResult {
  escalationType: string;
  previousState: {
    priority?: Priority;
    assignedTo?: string;
  };
  newState: {
    priority?: Priority;
    assignedTo?: string;
    incidentId?: string;
  };
}

// ===== LINK TO INCIDENT COMMAND =====

export interface LinkToIncidentCommand extends ICommand {
  readonly type: 'LinkToIncident';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    incidentId: string;
    linkType: 'trigger' | 'related' | 'evidence';
    linkedBy: string;
    notes?: string;
  };
}

export interface LinkToIncidentResult {
  incidentId: string;
  linkType: string;
  allIncidentLinks: string[];
}

// ===== BATCH CREATE ACTIVITIES COMMAND =====

export interface BatchCreateActivitiesCommand extends ICommand {
  readonly type: 'BatchCreateActivities';
  readonly data: {
    activities: Array<{
      activityType: ActivityType;
      title: string;
      location: string;
      priority: Priority;
      description?: string;
      building?: string;
      zone?: string;
      confidence?: number;
      externalData?: {
        sourceSystem: string;
        originalType: string;
        rawPayload: Record<string, any>;
      };
    }>;
    batchId: string;
    sourceSystem?: string;
  };
}

export interface BatchCreateActivitiesResult {
  batchId: string;
  successCount: number;
  failedCount: number;
  createdActivityIds: string[];
  failures: Array<{
    index: number;
    error: string;
  }>;
  incidentsTriggered: Array<{
    activityId: string;
    incidentId: string;
    rule: string;
  }>;
}

// ===== MERGE ACTIVITIES COMMAND =====

export interface MergeActivitiesCommand extends ICommand {
  readonly type: 'MergeActivities';
  readonly data: {
    primaryActivityId: string;
    activityIdsToMerge: string[];
    mergeReason: string;
    mergedBy: string;
    preserveData?: {
      tags: boolean;
      assignments: boolean;
      timeline: boolean;
    };
  };
}

export interface MergeActivitiesResult {
  primaryActivityId: string;
  mergedActivityIds: string[];
  mergedData: {
    tagsAdded: string[];
    timelineEntriesAdded: number;
    evidenceLinked: string[];
  };
}

// Union type of all activity commands
export type ActivityCommand = 
  | CreateActivityCommand
  | UpdateActivityCommand
  | AssignActivityCommand
  | ArchiveActivityCommand
  | BulkUpdateStatusCommand
  | AddUserTagCommand
  | RemoveUserTagCommand
  | EscalateActivityCommand
  | LinkToIncidentCommand
  | BatchCreateActivitiesCommand
  | MergeActivitiesCommand;