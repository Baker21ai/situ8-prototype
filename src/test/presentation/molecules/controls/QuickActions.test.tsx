import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { 
  QuickActions, 
  QuickActionsCompact,
  QuickActionsMinimal,
  QuickActionsFloating,
  CriticalQuickActions,
  ReviewQuickActions,
  QuickActionsProps 
} from '../../../../presentation/molecules/controls/QuickActions';
import { EnterpriseActivity, ActivityCluster } from '../../../../../lib/types/activity';

// Mock ActionButton
vi.mock('../../../../presentation/atoms/buttons/ActionButton', () => ({
  ActionButton: ({ children, onClick, disabled, variant, icon, size }: any) => (
    <button 
      data-testid="action-button"
      onClick={onClick}
      disabled={disabled}
      className={`action-button ${variant} ${size}`}
      data-variant={variant}
    >
      {icon}
      {children}
    </button>
  )
}));

// Mock UI components
vi.mock('../../../../../components/ui/button', () => ({
  Button: ({ children, onClick, disabled, size, variant, className, title, ...props }: any) => (
    <button 
      data-testid="button"
      onClick={onClick}
      disabled={disabled}
      className={`${size} ${variant} ${className}`}
      title={title}
      {...props}
    >
      {children}
    </button>
  )
}));

// Mock Lucide icons
vi.mock('lucide-react', () => ({
  AlertTriangle: () => <div data-testid="alert-triangle-icon" />,
  User: () => <div data-testid="user-icon" />,
  Eye: () => <div data-testid="eye-icon" />,
  FileText: () => <div data-testid="file-text-icon" />,
  Clock: () => <div data-testid="clock-icon" />,
  CheckCircle: () => <div data-testid="check-circle-icon" />,
  XCircle: () => <div data-testid="x-circle-icon" />,
  MoreHorizontal: () => <div data-testid="more-horizontal-icon" />,
  AlertCircle: () => <div data-testid="alert-circle-icon" />
}));

describe('QuickActions', () => {
  const mockActivity: EnterpriseActivity = {
    id: '1',
    title: 'Security Alert',
    description: 'Unauthorized access detected',
    type: 'security-breach',
    priority: 'high',
    status: 'detecting',
    timestamp: '2024-01-15T14:30:00Z',
    location: 'Building A',
    zone: 'Zone 1'
  };

  const mockCluster: ActivityCluster = {
    id: 'cluster-1',
    clusterType: 'cluster',
    title: '5 activities in Building A',
    description: 'Multiple security events detected',
    location: 'Building A',
    timestamp: '2024-01-15T14:30:00Z',
    lastActivity: '2024-01-15T15:00:00Z',
    count: 5,
    highestPriority: 'critical',
    types: ['security-breach', 'alert'],
    timeRange: '30 minutes',
    activities: [mockActivity]
  };

  const mockOnAction = vi.fn();

  const defaultProps: QuickActionsProps = {
    activity: mockActivity,
    onAction: mockOnAction
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe('Rendering', () => {
    it('should render with default props', () => {
      render(<QuickActions {...defaultProps} />);
      
      // Should show escalate (high priority can be escalated)
      expect(screen.getByTestId('alert-triangle-icon')).toBeInTheDocument();
      // Should show assign (no assignedTo)
      expect(screen.getByTestId('user-icon')).toBeInTheDocument();
      // Should show view
      expect(screen.getByTestId('eye-icon')).toBeInTheDocument();
    });

    it('should render different variants', () => {
      const { rerender } = render(<QuickActions {...defaultProps} variant="full" />);
      expect(screen.getByText('Escalate')).toBeInTheDocument();
      
      rerender(<QuickActions {...defaultProps} variant="compact" />);
      expect(screen.queryByText('Escalate')).not.toBeInTheDocument();
      
      rerender(<QuickActions {...defaultProps} variant="minimal" />);
      expect(screen.queryByText('Escalate')).not.toBeInTheDocument();
    });
  });

  describe('Action filtering by conditions', () => {
    it('should show escalate action for non-critical activities', () => {
      render(<QuickActions activity={mockActivity} />);
      
      expect(screen.getByTestId('alert-triangle-icon')).toBeInTheDocument();
    });

    it('should not show escalate action for critical activities', () => {
      const criticalActivity = { ...mockActivity, priority: 'critical' as const };
      render(<QuickActions activity={criticalActivity} />);
      
      const escalateButtons = screen.queryAllByTestId('action-button').filter(btn => 
        btn.querySelector('[data-testid="alert-triangle-icon"]')
      );
      expect(escalateButtons).toHaveLength(0);
    });

    it('should show assign action for unassigned activities', () => {
      render(<QuickActions activity={mockActivity} />);
      
      expect(screen.getByTestId('user-icon')).toBeInTheDocument();
    });

    it('should not show assign action for assigned activities', () => {
      const assignedActivity = { ...mockActivity, assignedTo: 'John Doe' };
      render(<QuickActions activity={assignedActivity} />);
      
      const assignButtons = screen.queryAllByTestId('action-button').filter(btn => 
        btn.querySelector('[data-testid="user-icon"]')
      );
      expect(assignButtons).toHaveLength(0);
    });

    it('should show mark resolved for non-resolved activities', () => {
      render(<QuickActions activity={mockActivity} />);
      
      expect(screen.getByTestId('check-circle-icon')).toBeInTheDocument();
    });

    it('should not show mark resolved for resolved activities', () => {
      const resolvedActivity = { ...mockActivity, status: 'resolved' as const };
      render(<QuickActions activity={resolvedActivity} />);
      
      const resolveButtons = screen.queryAllByTestId('action-button').filter(btn => 
        btn.querySelector('[data-testid="check-circle-icon"]')
      );
      expect(resolveButtons).toHaveLength(0);
    });

    it('should not show create incident and add to case for clusters', () => {
      render(<QuickActions activity={mockCluster} />);
      
      expect(screen.queryByTestId('alert-triangle-icon')).not.toBeInTheDocument();
      expect(screen.queryByTestId('file-text-icon')).not.toBeInTheDocument();
    });
  });

  describe('Available actions filtering', () => {
    it('should only show specified available actions', () => {
      render(
        <QuickActions 
          {...defaultProps} 
          availableActions={['view', 'escalate']}
        />
      );
      
      expect(screen.getByTestId('eye-icon')).toBeInTheDocument();
      expect(screen.getByTestId('alert-triangle-icon')).toBeInTheDocument();
      expect(screen.queryByTestId('user-icon')).not.toBeInTheDocument();
    });

    it('should show all actions when availableActions is not specified', () => {
      render(<QuickActions {...defaultProps} />);
      
      expect(screen.getByTestId('eye-icon')).toBeInTheDocument();
      expect(screen.getByTestId('alert-triangle-icon')).toBeInTheDocument();
      expect(screen.getByTestId('user-icon')).toBeInTheDocument();
    });
  });

  describe('Variant behaviors', () => {
    it('should limit actions in minimal variant', () => {
      render(<QuickActions {...defaultProps} variant="minimal" />);
      
      const actionButtons = screen.getAllByTestId('button');
      expect(actionButtons.length).toBeLessThanOrEqual(2);
    });

    it('should limit actions in compact variant', () => {
      render(<QuickActions {...defaultProps} variant="compact" />);
      
      const actionButtons = screen.getAllByTestId('action-button');
      expect(actionButtons.length).toBeLessThanOrEqual(4);
    });

    it('should limit actions in floating variant', () => {
      render(<QuickActions {...defaultProps} variant="floating" />);
      
      const actionButtons = screen.getAllByTestId('action-button');
      expect(actionButtons.length).toBeLessThanOrEqual(3);
    });

    it('should render floating variant with fixed positioning', () => {
      const { container } = render(<QuickActions {...defaultProps} variant="floating" />);
      
      const floatingContainer = container.firstChild as HTMLElement;
      expect(floatingContainer.className).toContain('fixed');
      expect(floatingContainer.className).toContain('bottom-4');
      expect(floatingContainer.className).toContain('right-4');
    });

    it('should show icons only in minimal variant', () => {
      render(<QuickActions {...defaultProps} variant="minimal" />);
      
      const buttons = screen.getAllByTestId('button');
      buttons.forEach(button => {
        expect(button.textContent).toBe(''); // No text, only icons
      });
    });
  });

  describe('Action handling', () => {
    it('should call onAction when escalate is clicked', async () => {
      const user = userEvent.setup();
      render(<QuickActions {...defaultProps} />);
      
      const escalateButton = screen.getAllByTestId('action-button').find(btn => 
        btn.querySelector('[data-testid="alert-triangle-icon"]')
      );
      
      await user.click(escalateButton!);
      expect(mockOnAction).toHaveBeenCalledWith('escalate', mockActivity);
    });

    it('should call onAction when view is clicked', async () => {
      const user = userEvent.setup();
      render(<QuickActions {...defaultProps} />);
      
      const viewButton = screen.getAllByTestId('action-button').find(btn => 
        btn.querySelector('[data-testid="eye-icon"]')
      );
      
      await user.click(viewButton!);
      expect(mockOnAction).toHaveBeenCalledWith('view', mockActivity);
    });

    it('should call onAction when assign is clicked', async () => {
      const user = userEvent.setup();
      render(<QuickActions {...defaultProps} />);
      
      const assignButton = screen.getAllByTestId('action-button').find(btn => 
        btn.querySelector('[data-testid="user-icon"]')
      );
      
      await user.click(assignButton!);
      expect(mockOnAction).toHaveBeenCalledWith('assign', mockActivity);
    });

    it('should call onAction when more options is clicked', async () => {
      const user = userEvent.setup();
      render(<QuickActions {...defaultProps} />);
      
      const moreButton = screen.getAllByTestId('button').find(btn => 
        btn.querySelector('[data-testid="more-horizontal-icon"]')
      );
      
      await user.click(moreButton!);
      expect(mockOnAction).toHaveBeenCalledWith('more_options', mockActivity);
    });
  });

  describe('Cluster-specific actions', () => {
    it('should show cluster-specific actions', () => {
      render(<QuickActions activity={mockCluster} />);
      
      expect(screen.getByText('View All 5')).toBeInTheDocument();
      expect(screen.getByText('Split Cluster')).toBeInTheDocument();
    });

    it('should call onAction for view all cluster action', async () => {
      const user = userEvent.setup();
      render(<QuickActions activity={mockCluster} onAction={mockOnAction} />);
      
      const viewAllButton = screen.getByText('View All 5');
      await user.click(viewAllButton);
      
      expect(mockOnAction).toHaveBeenCalledWith('view_all', mockCluster);
    });

    it('should call onAction for split cluster action', async () => {
      const user = userEvent.setup();
      render(<QuickActions activity={mockCluster} onAction={mockOnAction} />);
      
      const splitButton = screen.getByText('Split Cluster');
      await user.click(splitButton);
      
      expect(mockOnAction).toHaveBeenCalledWith('split_cluster', mockCluster);
    });
  });

  describe('Disabled state', () => {
    it('should disable all actions when disabled prop is true', () => {
      render(<QuickActions {...defaultProps} disabled={true} />);
      
      const buttons = screen.getAllByTestId('button');
      const actionButtons = screen.getAllByTestId('action-button');
      
      [...buttons, ...actionButtons].forEach(button => {
        expect(button).toBeDisabled();
      });
    });

    it('should not call onAction when disabled', async () => {
      const user = userEvent.setup();
      render(<QuickActions {...defaultProps} disabled={true} />);
      
      const viewButton = screen.getAllByTestId('action-button').find(btn => 
        btn.querySelector('[data-testid="eye-icon"]')
      );
      
      await user.click(viewButton!);
      expect(mockOnAction).not.toHaveBeenCalled();
    });
  });

  describe('Button sizing', () => {
    it('should use correct button sizes for variants', () => {
      const { rerender } = render(<QuickActions {...defaultProps} variant="full" />);
      
      let actionButtons = screen.getAllByTestId('action-button');
      expect(actionButtons[0].className).toContain('default');
      
      rerender(<QuickActions {...defaultProps} variant="compact" />);
      actionButtons = screen.getAllByTestId('action-button');
      expect(actionButtons[0].className).toContain('sm');
      
      rerender(<QuickActions {...defaultProps} variant="minimal" />);
      const buttons = screen.getAllByTestId('button');
      expect(buttons[0].className).toContain('sm');
    });
  });

  describe('Action variants', () => {
    it('should apply correct action variants', () => {
      render(<QuickActions {...defaultProps} />);
      
      const escalateButton = screen.getAllByTestId('action-button').find(btn => 
        btn.querySelector('[data-testid="alert-triangle-icon"]')
      );
      expect(escalateButton).toHaveAttribute('data-variant', 'critical');
      
      const assignButton = screen.getAllByTestId('action-button').find(btn => 
        btn.querySelector('[data-testid="user-icon"]')
      );
      expect(assignButton).toHaveAttribute('data-variant', 'secondary');
    });
  });

  describe('Tooltips in minimal variant', () => {
    it('should show tooltips for icons in minimal variant', () => {
      render(<QuickActions {...defaultProps} variant="minimal" />);
      
      const buttons = screen.getAllByTestId('button');
      buttons.forEach(button => {
        expect(button).toHaveAttribute('title');
      });
    });
  });

  describe('Specialized variants', () => {
    it('should render QuickActionsCompact correctly', () => {
      render(<QuickActionsCompact activity={mockActivity} />);
      
      const actionButtons = screen.getAllByTestId('action-button');
      expect(actionButtons[0].className).toContain('sm');
    });

    it('should render QuickActionsMinimal correctly', () => {
      render(<QuickActionsMinimal activity={mockActivity} />);
      
      const buttons = screen.getAllByTestId('button');
      expect(buttons.length).toBeLessThanOrEqual(2);
    });

    it('should render QuickActionsFloating correctly', () => {
      const { container } = render(<QuickActionsFloating activity={mockActivity} />);
      
      const floatingContainer = container.firstChild as HTMLElement;
      expect(floatingContainer.className).toContain('fixed');
      expect(floatingContainer.className).toContain('z-50');
    });

    it('should render CriticalQuickActions with specific actions', () => {
      render(<CriticalQuickActions activity={mockActivity} />);
      
      expect(screen.getByTestId('alert-triangle-icon')).toBeInTheDocument(); // escalate
      expect(screen.getByTestId('user-icon')).toBeInTheDocument(); // assign
      expect(screen.queryByTestId('check-circle-icon')).not.toBeInTheDocument(); // no mark resolved
    });

    it('should render ReviewQuickActions with specific actions', () => {
      render(<ReviewQuickActions activity={mockActivity} />);
      
      expect(screen.getByTestId('eye-icon')).toBeInTheDocument(); // view
      expect(screen.getByTestId('check-circle-icon')).toBeInTheDocument(); // mark resolved
      expect(screen.getByTestId('x-circle-icon')).toBeInTheDocument(); // dismiss
      expect(screen.queryByTestId('alert-triangle-icon')).not.toBeInTheDocument(); // no escalate
    });
  });

  describe('Full action workflow', () => {
    it('should show create incident action', () => {
      render(<QuickActions {...defaultProps} />);
      
      expect(screen.getByText('Create Incident')).toBeInTheDocument();
    });

    it('should show add to case action', () => {
      render(<QuickActions {...defaultProps} />);
      
      expect(screen.getByText('Add to Case')).toBeInTheDocument();
    });

    it('should show mark resolved action', () => {
      render(<QuickActions {...defaultProps} />);
      
      expect(screen.getByText('Mark Resolved')).toBeInTheDocument();
    });

    it('should show dismiss action', () => {
      render(<QuickActions {...defaultProps} />);
      
      expect(screen.getByText('Dismiss')).toBeInTheDocument();
    });
  });

  describe('Accessibility', () => {
    it('should have proper button roles', () => {
      render(<QuickActions {...defaultProps} />);
      
      const buttons = screen.getAllByRole('button');
      expect(buttons.length).toBeGreaterThan(0);
    });

    it('should be keyboard navigable', async () => {
      const user = userEvent.setup();
      render(<QuickActions {...defaultProps} />);
      
      const firstButton = screen.getAllByRole('button')[0];
      await user.tab();
      
      expect(firstButton).toHaveFocus();
    });

    it('should handle keyboard activation', async () => {
      const user = userEvent.setup();
      render(<QuickActions {...defaultProps} />);
      
      const viewButton = screen.getAllByTestId('action-button').find(btn => 
        btn.querySelector('[data-testid="eye-icon"]')
      );
      
      viewButton?.focus();
      await user.keyboard('{Enter}');
      
      expect(mockOnAction).toHaveBeenCalledWith('view', mockActivity);
    });
  });

  describe('Custom styling', () => {
    it('should apply custom className', () => {
      const customClass = 'custom-quick-actions';
      const { container } = render(<QuickActions {...defaultProps} className={customClass} />);
      
      expect(container.firstChild).toHaveClass(customClass);
    });
  });

  describe('Edge cases', () => {
    it('should handle activity without onAction callback', () => {
      render(<QuickActions activity={mockActivity} />);
      
      // Should render without errors
      expect(screen.getByTestId('eye-icon')).toBeInTheDocument();
    });

    it('should handle empty availableActions array', () => {
      render(<QuickActions {...defaultProps} availableActions={[]} />);
      
      // Should show only more options button
      const actionButtons = screen.queryAllByTestId('action-button');
      expect(actionButtons).toHaveLength(0);
    });
  });

  describe('Snapshot tests', () => {
    it('should match snapshot for full variant', () => {
      const { container } = render(<QuickActions activity={mockActivity} variant="full" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for minimal variant', () => {
      const { container } = render(<QuickActions activity={mockActivity} variant="minimal" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for floating variant', () => {
      const { container } = render(<QuickActions activity={mockActivity} variant="floating" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
