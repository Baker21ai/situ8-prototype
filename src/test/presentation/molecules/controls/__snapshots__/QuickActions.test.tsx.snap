// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuickActions > Snapshot tests > should match snapshot for floating variant 1`] = `
<div
  class="
        fixed bottom-4 right-4 z-50
        bg-white dark:bg-gray-800
        rounded-lg shadow-lg border
        p-2 flex gap-2
        
      "
>
  <button
    class="action-button critical default"
    data-testid="action-button"
    data-variant="critical"
  >
    <div
      data-testid="alert-triangle-icon"
    />
  </button>
  <button
    class="action-button secondary default"
    data-testid="action-button"
    data-variant="secondary"
  >
    <div
      data-testid="user-icon"
    />
  </button>
  <button
    class="action-button secondary default"
    data-testid="action-button"
    data-variant="secondary"
  >
    <div
      data-testid="eye-icon"
    />
  </button>
  <button
    class="default ghost min-w-0 h-8 w-8 p-0"
    data-testid="button"
  >
    <div
      data-testid="more-horizontal-icon"
    />
  </button>
</div>
`;

exports[`QuickActions > Snapshot tests > should match snapshot for full variant 1`] = `
<div
  class="flex items-center gap-2 flex-wrap "
>
  <div
    class="flex items-center gap-1"
  >
    <button
      class="action-button critical default"
      data-testid="action-button"
      data-variant="critical"
    >
      <div
        data-testid="alert-triangle-icon"
      />
      Escalate
    </button>
    <button
      class="action-button secondary default"
      data-testid="action-button"
      data-variant="secondary"
    >
      <div
        data-testid="user-icon"
      />
      Assign
    </button>
    <button
      class="action-button secondary default"
      data-testid="action-button"
      data-variant="secondary"
    >
      <div
        data-testid="eye-icon"
      />
      View
    </button>
  </div>
  <button
    class="default ghost h-8 w-8 p-0"
    data-testid="button"
  >
    <div
      data-testid="more-horizontal-icon"
    />
  </button>
</div>
`;

exports[`QuickActions > Snapshot tests > should match snapshot for minimal variant 1`] = `
<div
  class="flex items-center gap-1 "
>
  <button
    class="sm ghost h-6 w-6 p-0"
    data-testid="button"
    title="Escalate"
  >
    <div
      data-testid="alert-triangle-icon"
    />
  </button>
  <button
    class="sm ghost h-6 w-6 p-0"
    data-testid="button"
    title="Assign"
  >
    <div
      data-testid="user-icon"
    />
  </button>
</div>
`;
